backpropagation/network/neural_network.py

```diff
@@ -56,14 +61,15 @@ def _feedforward(self, x: np.array) -> np.array:
                 The output values of the neural network for the given input.
 
         """
-        for weight in self.weights:
-            x = self.activation_function.calculate_value(
-                weight @ np.insert(x, 0, 1)
-            )
+        calculate_activation = self.activation_function.calculate_value
```

Comment:
For loop was more readable here

Reply:
During our previous code review we pointed out that this is a classic example for using reduce.

Reply:
Okay.